promdump/promdump.go

```diff
@@ -651,9 +770,52 @@ func main() {
 		}
 	}
 	if *metric != "" {
-		err = exportMetric(ctx, promApi, *metric, beginTS, endTS, *periodDur, *batchDur, *batchesPerFile, *out)
+		customMetricCount, err = exportMetric(ctx, promApi, *metric, beginTS, endTS, *periodDur, *batchDur, *batchesPerFile, *out)
 		if err != nil {
 			log.Fatalln("exportMetric:", err)
 		}
 	}
+
+	if *enableTar {
+		var filename string
+		filename = *tarFilename
+		if *tarFilename == "" {
+			// If filename is not provided, generate a default tar filename
+			filename = generateDefaultTarFilename()
+		} else {
+			// Check if filename already exists
+			if _, err := os.Stat(filename); err == nil {
+				log.Fatalf("Error: %s already exists. Please choose a different filename.\n", filename)
```

Comment:
You'll usually want to default to using %v because %v will mostly do the right thing. When you use %v, the default output format for strings is %s.

I'd recommend changing this fatal to look like this:

```
"specified --tar_filename '%s' already exists; please choose a different filename"
```

I removed "Error:" from the beginning of this message because it's a fatal, not an error and because we mostly don't have our log messages tagged in this way. There are a few places where we have some inconsistencies in terms of prefixing errors and warnings, so we should clean that up in a future version.

Wrapping literals like filenames in quotes makes it easier to parse messages, for both humans and code. Most of the messages emitted by promdump follow this convention.